### PR TITLE
LibMarkdown: Make images shrink-to-fit, which you can toggle by clicking on them

### PR DESCRIPTION
**Pull request by AtkinsSJ**
Previously, they always showed at 100% size, which made viewing certain manual pages quite awkward. So, this PR makes them a bit smarter: If they are too wide, they shrink to fit the viewport width, with a "zoom" cursor. Clicking the image then toggles them being 100% size, or back to shrink-to-fit.

### Before clicking
image

### After clicking
image

### CODE REVIEW

File: Userland/Libraries/LibWeb/Page/EventHandler.cpp [hunk 1]
```diff
@@ -75,6 +75,10 @@ static Gfx::StandardCursor cursor_css_to_gfx(Optional<CSS::Cursor> cursor)
     case CSS::Cursor::NwseResize:
         return Gfx::StandardCursor::ResizeDiagonalTLBR;
 
+    case CSS::Cursor::ZoomIn:
+    case CSS::Cursor::ZoomOut:
+        return Gfx::StandardCursor::Zoom;
+
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Page/EventHandler.cpp:81]: Do me a favor and get rid of the newlines from line 53 onwards instead, who did this lol

File: Userland/Libraries/LibMarkdown/Document.cpp [hunk 2]
```diff
@@ -15,20 +15,60 @@ namespace Markdown {
 String Document::render_to_html() const
 {
     StringBuilder builder;
+    builder.append(R"~~~(<!DOCTYPE html>
```
Comment (linusg) [inline Userland/Libraries/LibMarkdown/Document.cpp:18]: is `std::embed` a thing yet? 😅

File: Userland/Libraries/LibMarkdown/Document.cpp [hunk 3]
```diff
@@ -15,20 +15,60 @@ namespace Markdown {
...
+        function processImages() {
+            let images = document.querySelectorAll("img");
+            let windowWidth = window.innerWidth;
+            images.forEach((image) => {
+                if (image.naturalWidth > windowWidth) {
+                    image.classList.add("zoomable");
+                } else {
+                    image.classList.remove("zoomable");
+                    image.classList.remove("zoomed-in");
+                }
```
Comment (linusg) [inline Userland/Libraries/LibMarkdown/Document.cpp:48]: Not a huge fan of outputting this into _any_ HTML generated from markdown... shrink-to-fit by default + having the user of the HTML insert a script into the webview that adds such customisation seems cleaner

Comment (linusg) [reply Userland/Libraries/LibMarkdown/Document.cpp:48]: _Or_ a flag to optionally enable features such as zoomable images, code highlighting, etc which can be passed by Help, TextEditor, and whoever else is interested in a rich HTML document.
Ok, I think we have enough options 😄

Comment (AtkinsSJ) [reply Userland/Libraries/LibMarkdown/Document.cpp:48]: You're right, it's a bit gross given it's a user-side feature. Thank you for keeping my hack tendencies under control. :wink:

**Comment by AtkinsSJ** [comment]
New changes:
- Add a parameter to `Markdown::Document::render_to_html()` to provide extra contents for the HTML `<head>` element.
- Use that to insert the extra CSS/JS for the shrink-to-fit images, instead of LibMarkdown always inserting it.
- Also, use that for HackStudio documentation tooltips. (I couldn't test this however, because I can't figure out how to make the tooltips appear. :sweat_smile: I think they might just be broken. Poor HackStudio.)
- Get Welcome to render the README Markdown using LibWeb instead of manually. (Which means it gets the shrink-to-fit images too.)
- And most importantly: Got rid of the extra newlines. :^)

**Review by linusg — APPROVED** [review]
Nice!

**Pull request merged**